docs/source/checks/tabular/integrity/plot_mixed_data_types.py

```diff
@@ -2,27 +2,79 @@
 """
 Mixed Data Types
 ****************
-"""
 
-#%%
+This notebooks provides an overview for using and understanding the mixed data types check.
 
-import pandas as pd
+**Structure:**
+
+* `What are Mixed Data Types? <#what-are-mixed-data-types>`__
+* `Run the Check <#run-the-check>`__
+* `Define a Condition <#define-a-condition>`__
+
+What are Mixed Data Types?
+==========================
+Mixed data types is a situation when a column contains both string values and numeric values (either as numeric type or
+as string like "42.90"). The check detects columns which contains mixed types at any ratio, even if there is a single
+value which is different from the rest.
 
+We want to be aware of this since it may indicate a problem in the data collection pipeline, or represent a
+problem for the model's training.
+
+Run the Check
+=============
+We will run the check on the adult dataset which can be downloaded from the
+`UCI machine learning repository <http://archive.ics.uci.edu/ml>`_ and is also available in
+`deepchecks.tabular.datasets`, and introduce to it some data type mixing in order to show the check's result.
+"""
+
+import pandas as pd
+import numpy as np
+from deepchecks.tabular import Dataset
+from deepchecks.tabular.datasets.classification import adult
 from deepchecks.tabular.checks import MixedDataTypes
 
-#%%
-# Benign condition
-# ================
 
-data = {'col1': ['foo', 'bar', 'cat']}
-dataframe = pd.DataFrame(data=data)
-MixedDataTypes().add_condition_rare_type_ratio_not_in_range().run(dataframe)
+def insert_new_values_types(col: pd.Series, ratio_to_replace: float, values_list):
+    col = col.to_numpy().astype(object)
+    indices_to_replace = np.random.choice(range(len(col)), int(len(col) * ratio_to_replace), replace=False)
+    new_values = np.random.choice(values_list, len(indices_to_replace))
+    col[indices_to_replace] = new_values
+    return col
+
+
+def insert_string_types(col: pd.Series, ratio_to_replace):
+    return insert_new_values_types(col, ratio_to_replace, ['a', 'b', 'c'])
+
+
+def insert_numeric_string_types(col: pd.Series, ratio_to_replace):
+    return insert_new_values_types(col, ratio_to_replace, ['1.0', '1', '10394.33'])
+
+
+def insert_number_types(col: pd.Series, ratio_to_replace):
+    return insert_new_values_types(col, ratio_to_replace, [66, 99.9])
+
+
+# Load dataset and insert some data type mixing
+adult_df, _ = adult.load_data(as_train_test=True, data_format='Dataframe')
+adult_df['workclass'] = insert_numeric_string_types(adult_df['workclass'], ratio_to_replace=0.01)
+adult_df['education'] = insert_number_types(adult_df['education'], ratio_to_replace=0.1)
+adult_df['age'] = insert_string_types(adult_df['age'], ratio_to_replace=0.5)
+
+# Run the check
+adult_dataset = Dataset(adult_df, cat_features=['workclass', 'education'])
+check = MixedDataTypes()
+result = check.run(adult_dataset)
+result
```

Comment:
split the part where you dirty the data and the part where you actually run the check to two different cells, for clarity and ease of copy-paste (Include relevant imports in the cell...)

Reply:
Doesn't splitting make it harder to copy paste? (I guess we want the copy paste to reproduce the notebook outcome)